tests/components/upnp/test_init.py

```diff
@@ -20,35 +27,104 @@
 async def test_async_setup_entry_default(hass):
     """Test async_setup_entry."""
     udn = "uuid:device_1"
+    location = "http://192.168.1.1/desc.xml"
     mock_device = MockDevice(udn)
-    discovery_infos = [
+    discoveries = [
         {
-            DISCOVERY_UDN: mock_device.udn,
+            DISCOVERY_LOCATION: location,
+            DISCOVERY_NAME: mock_device.name,
             DISCOVERY_ST: mock_device.device_type,
-            DISCOVERY_LOCATION: "http://192.168.1.1/desc.xml",
+            DISCOVERY_UDN: mock_device.udn,
+            DISCOVERY_UNIQUE_ID: mock_device.unique_id,
+            DISCOVERY_USN: mock_device.usn,
         }
     ]
     entry = MockConfigEntry(
-        domain=upnp.DOMAIN, data={"udn": mock_device.udn, "st": mock_device.device_type}
+        domain=upnp.DOMAIN,
+        data={
+            CONFIG_ENTRY_UDN: mock_device.udn,
+            CONFIG_ENTRY_ST: mock_device.device_type,
+        },
     )
 
     config = {
         # no upnp
     }
-    async_discover = AsyncMock(return_value=[])
+    async_discover = AsyncMock()
     with patch.object(
         Device, "async_create_device", AsyncMock(return_value=mock_device)
     ), patch.object(Device, "async_discover", async_discover):
         # initialisation of component, no device discovered
+        async_discover.return_value = []
+        await async_setup_component(hass, "upnp", config)
+        await hass.async_block_till_done()
+
+        # loading of config_entry, device discovered
+        async_discover.return_value = discoveries
+        assert await upnp.async_setup_entry(hass, entry) is True
```

Comment:
Please only use the core interfaces to start the tests. Here use `hass.config_entries.async_setup`. Although that shouldn't be needed since `async_setup_component` will set up the integration too.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations